### PR TITLE
Migrate spec based install components to operator managed components

### PR DESCRIPTION
**Pull request by piyush-nimbalkar**
**What this PR does / why we need it**:
- Pausing any component from getting installed until all the Portworx pods have been migrated. This is to ensure if there is any failure during Portworx pod migration, then there are less pieces to revert. Not all components are paused though. Components like PSP, auth, PX rbac and service and installed to not hinder Portworx functionality.

- Once the Portworx pod migration is completed, just deleting the old components and letting operator take care of creating the new ones.

- Still working on the unit tests and functional tests. Will add them to this PR as separate commits.

**Which issue(s) this PR fixes** (optional)
Closes # https://portworx.atlassian.net/browse/OPERATOR-458

**Special notes for your reviewer**:

### CODE REVIEW

**Review by pure-jliao — COMMENTED** [review]
so far looks pretty good to me.

File: pkg/migration/cleanup.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+		return err
+	}
+	if err := h.deletePortworxRbac(cluster); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (h *Handler) deletePortworxRbac(cluster *corev1.StorageCluster) error {
+	if err := h.deleteObject("node-role-binding", "", &rbacv1.ClusterRoleBinding{}, cluster); err != nil {
```
Comment (haibinxie) [inline pkg/migration/cleanup.go:44]: nit: please use const for the object names in the function.

File: pkg/migration/cleanup.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+	if err := h.deleteObject("node-get-put-list-role", "", &rbacv1.ClusterRole{}, cluster); err != nil {
+		return err
+	}
+	if err := h.deleteObject("px-role-binding", cluster.Namespace, &rbacv1.RoleBinding{}, cluster); err != nil {
+		return err
+	}
+	if err := h.deleteObject("px-role", cluster.Namespace, &rbacv1.Role{}, cluster); err != nil {
+		return err
+	}
+	if err := h.deleteObject("px-role-binding", "portworx", &rbacv1.RoleBinding{}, cluster); err != nil {
```
Comment (haibinxie) [inline pkg/migration/cleanup.go:56]: namespace is always "portworx"?

File: pkg/migration/generate.go [hunk 3]
```diff
@@ -2,23 +2,390 @@ package migration
...
+			envMap["PORTWORX_AUTH_SYSTEM_KEY"] = &v1.EnvVar{
+				Name:  "PORTWORX_AUTH_SYSTEM_KEY",
+				Value: c.Args[i+1],
+			}
+			i++
+		} else if arg == "--log" {
+			cluster.Annotations[pxutil.AnnotationLogFile] = c.Args[i+1]
+			i++
+		} else {
+			miscArgs += arg + " "
```
Comment (haibinxie) [inline pkg/migration/generate.go:293]: Just curious, due to the big list above, if we miss any parameter above, is there any pattern/way to cross check if this arg should belong to miscArgs?

**Review by haibinxie — APPROVED** [review]
Looks good

File: pkg/migration/cleanup.go (hunk 2 above)
Comment (piyush-nimbalkar) [reply pkg/migration/cleanup.go:56]: Yes, this is the default namespace we had for storing volume encryption secrets. Our spec generator still uses that namespace by default.

File: pkg/migration/generate.go (hunk 3 above)
Comment (piyush-nimbalkar) [reply pkg/migration/generate.go:293]: There is no pattern as such. If an argument does not have corresponding field in StorageCluster spec, we are adding it to the misc args. Misc args will take anything and pass it verbatim to oci-mon. At least, we won't change or break the existing PX deployment.
Saying that when we add new corresponding fields, we will have to keep this convert function up to date for better migration.

**Review by pure-jliao — APPROVED** [review]
LGTM!

**Pull request merged**